Remove ERC165Storage (#3880)
OpenZeppelin/openzeppelin-contracts: CHANGELOG.md

 * `TransparentUpgradeableProxy`: Removed `admin` and `implementation` getters, which were only callable by the proxy owner and thus not very useful. ([#3820](https://github.com/OpenZeppelin/openzeppelin-contracts/pull/3820))
 * `ProxyAdmin`: Removed `getProxyAdmin` and `getProxyImplementation` getters. ([#3820](https://github.com/OpenZeppelin/openzeppelin-contracts/pull/3820))
 * `ERC20`: Deleted `_beforeTokenTransfer` and `_afterTokenTransfer` hooks, added a new internal `_update` function for customizations, and refactored all extensions using those hooks to use `_update` instead. ([#3838](https://github.com/OpenZeppelin/openzeppelin-contracts/pull/3838))

>  * `ERC165Storage`: contract was removed. ([#](https://github.com/OpenZeppelin/openzeppelin-contracts/pull/))

Reviewer: ```suggestion
 * `ERC165Storage`: Removed this contract in favor of inheritance based approach. ([#3880](https://github.com/OpenZeppelin/openzeppelin-contracts/pull/3880))
```

  }
```

> #### ERC165Storage

Reviewer: ```suggestion
#### ERC165Storage

```

> As a result of removing the ERC165Storage contracts, users wanting to indicate their support for interfaces should do so by overriding the `supportsInterface` function as seen below:

Reviewer: This isn't bad but here is what I would write:

```suggestion
Users that were registering EIP-165 interfaces with `_registerInterface` from `ERC165Storage` should instead do so so by overriding the `supportsInterface` function as seen below:
```

> ```

Reviewer: ```suggestion
```solidity
```

>  function supportsInterface(bytes4 interfaceId) public view virtual override returns (bool) {
>     return interfaceId == type(MyInterface).interfaceId || super.supportsInterface(interfaceId);
>  }

Reviewer: ```suggestion
function supportsInterface(bytes4 interfaceId) public view virtual override returns (bool) {
    return interfaceId == type(MyInterface).interfaceId || super.supportsInterface(interfaceId);
}
```
